src/main/java/org/acme/infinispanclient/service/DataLoader.java

```diff
@@ -0,0 +1,113 @@
+package org.acme.infinispanclient.service;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+
+import org.acme.infinispanclient.model.HPCharacter;
+import org.acme.infinispanclient.model.HPMagic;
+import org.acme.infinispanclient.model.HPSpell;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.quarkus.runtime.StartupEvent;
+
+@ApplicationScoped
+public class DataLoader {
+   public static final String HP_CHARACTERS_NAME = "characters";
+   public static final String HP_SPELLS_NAME = "spells";
+   public static final String HP_MAGIC_NAME = "magic";
+
+   private static final Logger LOGGER = LoggerFactory.getLogger(DataLoader.class.getName());
+
+   @Inject
+   @ConfigProperty(name = "characters.filename")
+   private String charactersFileName;
+
+   @Inject
+   @ConfigProperty(name = "spells.filename")
+   private String spellsFileName;
+
+   @Inject
+   private RemoteCacheManager cacheManager;
+
+   void onStart(@Observes StartupEvent ev) {
+      LOGGER.info("On start - clean and load");
+      // Get or create caches
+      RemoteCache<Integer, HPCharacter> characters = cacheManager.administration().getOrCreateCache(HP_CHARACTERS_NAME, "default");
```

Comment:
Just a side note, but this something I wanted to improve at some point. That the user can define a cache configuration that we automatically register if the cache isn't present. Currently the assumption is that the cache is already configured on the server, which requires code like this.

Reply:
ok